extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/logical/StageMaker.java

```diff
@@ -91,13 +95,8 @@ private StageDefinitionBuilder buildStageInternal(LogicalStage stage)
   private StageDefinitionBuilder buildFrameProcessorStage(AbstractFrameProcessorStage frameProcessorStage)
   {
     List<LogicalInputSpec> inputs = frameProcessorStage.getInputSpecs();
-    List<InputSpec> inputSpecs = new ArrayList<>();
-    for (LogicalInputSpec dagInputSpec : inputs) {
-      inputSpecs.add(dagInputSpec.toInputSpec(this));
-    }
+    StageDefinitionBuilder sdb = newStageDefinitionBuilder(inputs);
     StageProcessor<?, ?> stageProcessor = frameProcessorStage.buildStageProcessor(this);
-    StageDefinitionBuilder sdb = newStageDefinitionBuilder();
-    sdb.inputs(inputSpecs);
     sdb.signature(frameProcessorStage.getLogicalRowSignature());
     sdb.processor(stageProcessor);
     sdb.shuffleSpec(MixShuffleSpec.instance());
```

Comment:
the Mix spec takes everything down to a single partition, which we want to avoid if there might be a lot of data. Another option is to just have _no_ shuffle spec, which means the output partitions are the same as the input partitions. In principle stages with no shuffle spec could be folded into the next stage, although that doesn't happen yet.

Anyway, maybe that makes more sense for the SegmentMapStage?